### PR TITLE
feat: add transfer command to migrate agent files between coding assistants

### LINKED ISSUE #63: feat: add transfer command to migrate agent files between coding assistants
## Summary

Add a `code-minions transfer` command that migrates installed agent and skill files from one coding assistant's directory layout to another (e.g. GitHub Copilot → Claude, Claude → Opencode, etc.).

## Problem

Users who have already installed code-minions packages for one coding assistant (e.g. GitHub Copilot) may want to switch to or additionally support a different assistant (e.g. Claude or Opencode). Currently, there's no way to do this without manually moving files between the assistant-specific directory layouts:

| Assistant | Agent Dir | Skill Dir |
|-----------|-----------|-----------|
| GitHub Copilot | `.github/agents/` | `skills/` |
| Claude | `.claude/agents/` | `.claude/skills/` |
| Opencode | `.opencode/agents/` | `.opencode/skills/` |

This means users have to understand each assistant's layout, manually copy/move files, and hope they don't miss anything. This is error-prone and tedious.

## Supported Transfer Paths

All permutations should be supported:

- GitHub Copilot → Claude
- GitHub Copilot → Opencode
- Claude → Opencode
- Claude → GitHub Copilot
- Opencode → GitHub Copilot
- Opencode → Claude

## Proposed CLI Interface

```bash
# Transfer all agent/skill files from Copilot layout to Claude layout
code-minions transfer --from copilot --to claude

# Dry run to preview what would be moved
code-minions transfer --from copilot --to claude --dry-run

# Force overwrite if destination files already exist
code-minions transfer --from copilot --to claude --force

# Transfer within a specific target directory
code-minions transfer --from claude --to opencode --target ./my-project
```

## Proposed Behaviour

1. **Discover** — Scan the source assistant's directories (e.g. `.github/agents/`, `skills/`) for installed agent and skill files
2. **Map** — Reverse-map source paths to the generic package layout (`agents/`, `skills/`), then forward-map to the destination assistant's layout
3. **Copy** — Write files to the destination layout
4. **Report** — Print a summary of copied, skipped, and errored files (consistent with `install`/`uninstall` output)

### Open Questions

- **Move vs Copy?** Should transfer *move* files (removing from source) or *copy* them (keeping both)? A `--move` flag could support both. Default to copy.
- **Cleanup source?** If moving, should empty source directories be cleaned up (like `uninstall` does)?
- **AGENTS.md handling?** Should the transfer also create/update the `AGENTS.md` file in the destination assistant's expected location?
- **Partial transfers?** Should users be able to transfer a single package (`--package developer-mentor`) or only all installed files?
- **Validation?** Should the command warn if the source directory doesn't exist or has no files?

## Implementation Notes

- The `assistant.Config` in `internal/assistant/assistant.go` already defines directory layouts for all three assistants and provides `NewPathMapper()` for forward mapping
- A **reverse path mapper** will be needed to go from an assistant-specific layout back to the generic layout (e.g. `.github/agents/foo.md` → `agents/foo.md`)
- The `Installer` struct in `internal/installer/installer.go` can likely be reused for the write side, but the read side will need to scan the local filesystem rather than the embedded FS
- New command in `internal/cmd/transfer.go`, wired into `root.go`
- Unit tests should cover all 6 transfer permutations, path edge cases, and dry-run/force flags

## Acceptance Criteria

- [ ] `code-minions transfer --from <assistant> --to <assistant>` copies agent and skill files between layouts
- [ ] All 6 assistant permutations are supported
- [ ] `--dry-run` previews changes without writing
- [ ] `--force` overwrites existing files at the destination
- [ ] `--target` specifies the working directory (defaults to `.`)
- [ ] Output summary matches the style of `install`/`uninstall` commands
- [ ] Unit tests cover all permutations and edge cases
- [ ] Command is documented in `--help` output

### PR DESCRIPTION
**Pull request by willvelida**
## Summary

Adds a \code-minions transfer\ command that copies agent and skill files from one coding assistant's directory layout to another (#63).

\\\ash
code-minions transfer --from copilot --to claude
\\\

## What it does

- **Copies** agent/skill files between assistant layouts (copilot, claude, opencode)
- **Translates** MCP server configurations between source and target formats (best-effort)
- **Regenerates** \AGENTS.md\ in the target layout (not copied, since it contains assistant-specific paths)
- Source files are left in place by default; use \--cleanup\ to delete them after a successful copy

## Flags

| Flag | Description |
|------|-------------|
| \--from\ | Source assistant (required) |
| \--to\ | Target assistant (required) |
| \--target\ | Working directory (default \.\) |
| \--dry-run\ | Preview without writing |
| \--force\ | Overwrite existing files |
| \--cleanup\ | Delete source files after successful copy |

All output modes supported: normal, verbose, quiet, JSON.

### Phase 1 — Reverse Path Mapper
- \NewReversePathMapper()\ on \ssistant.Config\ maps assistant-specific paths back to the generic layout
- Round-trip tested: forward → reverse = identity for all 3 assistants

### Phase 2 — Transfer Engine
- \installer.Transfer()\ scans source dirs with \ilepath.WalkDir\, reverse-maps → forward-maps → copies
- Skips metadata files (\package.yaml\, \mcp.yaml\, \AGENTS.md\)
- \--cleanup\ only deletes successfully copied files + empty source dirs
- Source validation: warns per missing dir, errors if both absent

### Phase 3 — CLI Command
- \
ewTransferCommand()\ wired into root command
- Orchestrates: \Transfer()\ → \mcp.Translate()\ → \AgentsMDHandler.OnInstall()\

### Phase 4 — Documentation
- README updated with usage examples, flags table, JSON example
- CHANGELOG entry added

## Tests

- **12 reverse mapper tests** (all 3 assistants + round-trip)
- **21 transfer engine tests** (all 6 permutations, force, skip, dry-run, cleanup, validation, metadata skip, nested dirs)
- **12 CLI command tests** (flag validation, integration, JSON, quiet, dry-run, force, cleanup, missing source)

Closes #63

### CODE REVIEW

File: internal/installer/transfer.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+			// Clean up empty source directories (deepest first)
+			cleanTransferDirs(sourceDirPaths)
+		}
+	}
+
+	return result, nil
+}
+
+// cleanEmptyDirs removes empty directories from deepest to shallowest.
```
[the remark this answers is not in the document]
Comment (willvelida) [reply internal/installer/transfer.go]: Fixed in ca8684d — updated the doc comment to reference `cleanTransferDirs`.

File: internal/cmd/transfer.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+			})
+			if err != nil {
+				// MCP translation failure is non-fatal — add as warning
+				transferResult.Warnings = append(transferResult.Warnings,
+					fmt.Sprintf("MCP config translation skipped: %v", err))
+				mcpResult = nil
+			}
+
+			// Step 3: Regenerate AGENTS.md in target layout
+			toCfg, _ := assistant.Get(to)
```
[the remark this answers is not in the document]
Comment (willvelida) [reply internal/cmd/transfer.go]: Fixed in ca8684d — now checking the error and returning early if it fails, consistent with the pattern in install.go.

File: internal/cmd/transfer.go [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+				Stdin:  os.Stdin,
+				Stdout: handlerStdout,
+			}
+
+			action, agentsMDErr := handler.OnInstall(agentsMDPath, []byte(installer.DefaultAgentsMDContent))
+			if agentsMDErr != nil {
+				transferResult.Errors = append(transferResult.Errors, fmt.Sprintf("AGENTS.md: %v", agentsMDErr))
+			} else if action == "created" {
+				transferResult.Copied = append(transferResult.Copied, agentsMDPath)
+			}
```
[the remark this answers is not in the document]
Comment (willvelida) [reply internal/cmd/transfer.go]: Fixed in 6ceb9ad — went with option (a): the transfer command now removes the existing AGENTS.md before calling OnInstall, so it's always written fresh. This avoids stale routing paths when transferring to a layout that already has an AGENTS.md from a previous assistant.

File: internal/installer/transfer.go [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+// transferSkipFiles lists filenames that should not be transferred.
+// AGENTS.md is excluded because it is regenerated for the target layout.
+var transferSkipFiles = map[string]bool{
+	"package.yaml": true,
+	"mcp.yaml":     true,
+	"AGENTS.md":    true,
+}
+
```
[the remark this answers is not in the document]
Comment (willvelida) [reply internal/installer/transfer.go]: Fixed in 6ceb9ad — transferSkipFiles is now derived from the shared SkipFiles map via an init function that copies all entries and adds AGENTS.md. No more duplicate maintenance.
